sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -78,6 +79,26 @@ final class DataFrameWriter private[sql](df: DataFrame) {
   }
 
   /**
+   * Set the trigger interval for the stream query.
+   *
+   * @since 2.0.0
+   */
+  def trigger(interval: Duration): DataFrameWriter = {
+    trigger = ProcessingTime(interval.toMillis)
+    this
+  }
+
+  /**
+   * Set the trigger interval for the stream query.
+   *
+   * @since 2.0.0
+   */
+  def trigger(interval: Long, unit: TimeUnit): DataFrameWriter = {
```

Comment:
Not all trigger modes are going to be time based though.  In the doc we also propose data sized based triggers.